ql/src/java/org/apache/hadoop/hive/ql/plan/DropFunctionDesc.java

```diff
@@ -61,4 +65,14 @@ public void setTemp(boolean isTemp) {
     this.isTemp = isTemp;
   }
 
+  /**
+   * @return what kind of replication scope this create is running under.
+   * This can result in a "DROP IF NEWER THAN" kind of semantic
+   */
+  public ReplicationSpec getReplicationSpec() {
+    if (replicationSpec == null) {
```

Comment:
The null check is to handle the non-repl-load case where they directly dereference the replSpec  returned. If remove null check here, then need to add one in the caller. But, I prefer to keep it here.